app/src/main/java/com/androidapp/taskmaster/adapters/TasksRecycleViewAdapter.java

```diff
@@ -0,0 +1,66 @@
+package com.androidapp.taskmaster.adapters;
+
+import android.content.Context;
+import android.content.Intent;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.androidapp.taskmaster.MainActivity;
+import com.androidapp.taskmaster.R;
+import com.androidapp.taskmaster.activities.TaskDetailsActivity;
+import com.androidapp.taskmaster.models.Task;
+
+import java.util.List;
+
+public class TasksRecycleViewAdapter extends RecyclerView.Adapter<TasksRecycleViewAdapter.TasksListViewHolder> {
+
+    List<Task> tasks;
+    Context callingActivity;
+
+    public TasksRecycleViewAdapter(List<Task> tasks, Context callingActivity) {
+        this.tasks = tasks;
+        this.callingActivity = callingActivity;
+    }
+
+    @NonNull
+    @Override
+    public TasksListViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View taskFragment = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.fragment_task_list, parent, false);
+        return new TasksListViewHolder(taskFragment);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull TasksListViewHolder holder, int position) {
+        TextView taskFragmentTextView = (TextView) holder.itemView.findViewById(R.id.taskfragmentTextView);
+        String taskTitle = tasks.get(position).getTitle();
+        String taskFragmentText = position + "." + taskTitle;
+        taskFragmentTextView.setText(taskFragmentText);
+
+        //Create on-click listener
+        taskFragmentTextView.setOnClickListener(v -> {
+            Intent goToTaskDetailsIntent = new Intent(callingActivity, TaskDetailsActivity.class);
+            goToTaskDetailsIntent.putExtra(MainActivity.TASK_NAME_TAG, taskTitle);
```

Comment:
Make sure the extra lists the activity that you're going to, not the one you're coming from (i.e., `goToTaskDetailsIntent.putExtra(TaskDetailsActivity.TASK_NAME_TAG, taskTitle)`)

Note: this will look in the TaskDetailsActivity for a property called TASK_NAME_TAG, so it can set the value of that property to taskTitle. If it doesn't find that property, the title doesn't get passed in.